test/run_test.py

```diff
@@ -0,0 +1,221 @@
+from retrace import *
+
+#Create repo with packages passed in arg packages
+def create_repo(packages, releaseid, crash_package):
+    print "Creating repo"
+    conf = config.Config()
+
+    repo_path = os.path.join(conf["RepoDir"], releaseid)
+    if not os.path.isdir(repo_path):
+        os.mkdir(repo_path)
+
+    #TODO if dnf exist, use that, else use yum
+    cmd = ["yum-deprecated", "--downloadonly", "--installroot=%s" % repo_path,
+           "--downloaddir=%s" % repo_path,
+           "install", "abrt-addon-ccpp", "shadow-utils", "gdb", "rpm",
+           "python", "gcore", "coreutils", crash_package]
+    #add all missing packages
+    for package in packages:
+        cmd.append(package[0])
+
+    child = Popen(cmd, stdout=PIPE, stderr=PIPE)
+    stdout, stderr = child.communicate()
+    if child.returncode:
+        print "ERROR: during downloading packages this errors occurred:"
+        print stderr
+
+    #TODO should use --installroot, but cause problems then
+    cmd = ["debuginfo-install", "--downloadonly",
+           "--downloaddir=%s" % repo_path, "abrt-addon-ccpp", "shadow-utils",
+           "gdb", "rpm", "gcc", "libffi", "python", crash_package]
+
+    child = Popen(cmd, stdout=PIPE, stderr=PIPE)
+    stdout, stderr = child.communicate()
+    if child.returncode:
+        print "ERROR: during downloading packages this errors occurred:"
+        print stderr
+
+    #create repo from downloaded packages
+    createrepo_cmd = ["createrepo", repo_path]
+    child = Popen(createrepo_cmd, stdout=PIPE, stderr=PIPE)
+    stdout, stderr = child.communicate()
+    if stderr:
+        print "ERROR: during creating repo this errors occurred:"
+        print stderr
+
+#Remove repo
+#Note: deletes repo even though, if was not created by this script
+def delete_repo(releaseid):
+    print "Deleting repo"
+    conf = config.Config()
+    repo_path = os.path.join(conf["RepoDir"], releaseid)
+    if not os.path.isdir(repo_path):
+        return
+    else:
+        shutil.rmtree(repo_path)
+
+#create coredump
+def generate_coredump():
+    print "Creating coredump"
+    #use gcore on this process
+    gcore_cmd = ['gcore', '-o', "/var/tmp/coredump", str(os.getpid())]
+    child = Popen(gcore_cmd, stdout=PIPE, stderr=PIPE)
+    stdout, stderr = child.communicate()
+    if child.returncode:
+        print "ERROR: following error was generated by gcore"
+        print stderr
+
+    #get only last line form output - there is filename mentioned
+    lastline = stdout.splitlines()[-1]
+    #get the last word, that should be coredump path
+    corepath = lastline[lastline.rfind(" ")+1:]
+    if not corepath.startswith("/var/tmp"):
+        print "ERROR: corefile was not generated"
+        sys.exit(1)
+
+    #rename it, so it is called coredump, not coredump.XXXXX
+    os.rename(corepath, corepath[:corepath.find(".")])
+    corepath = corepath[:corepath.find(".")]
+
+    #find in which package is python
+    rpm_cmdline = ['rpm', '-q', 'python']
```

Comment:
Oops, the correct path is `/proc/self/exe` and you must use `os.readlink("/proc/sefl/exe)` because the file is a symbolic link to the executable.